packages/docs/docs/docs/migrate_from_1.x_to_2.x.md

```diff
@@ -0,0 +1,117 @@
+---
+id: migrate_from_1.x_to_2.x
+title: Migrating to 2.x
+---
+
+In this section, you'll find everything you need to know if you're coming from ZeppelinOS 1.x and want to start using 2.x.
+
+## Terminology and fundamentals
+
+2.x's most fundamental change is that it expands on zOS's linking features. In 1.x, you were able to link your projects to what used to be called an "on-chain standard library". This "stdlib" was basically a pre-deployed version of OpenZeppelin, which gave your projects instant access to hundreds of standard implementations without having to compile or deploy any code whatsoever. This proved to be a very powerful, innovative concept, but we realized that it could be taken much further... In 1.x, you could connect to different versions of this "standard library" but there was just this ONE "standard library" that your application could link to. 2.x takes this to a whole new level; you can now connect your project to multiple "standard libraries", but that's not all, this makes the entire concept of what used to be a "standard library" evolve to the concept of an "EVM package".
+
+What is an EVM package? It is basically a piece of reusable code that has been deployed to the blockchain in the form of EVM bytecode. Anyone can create an EVM package, and your zOS projects will be able to connect to a vast ecosystem of pre-compiled, pre-deployed reusable code in the form of EVM packages.
+
+Similarly, as we realized what an EVM package was, we realized what it's not. In 1.x, your zOS project's contracts were always managed by two contracts: App and Package. This made your contracts upgradeable and gave them the ability to link to EVM packages (see [Contract architecture](https://docs.zeppelinos.org/docs/architecture.html) for more info). We realized that these two contracts made sense when it came to EVM packages, but didn't when it came to parts of your project that weren't meant to be reusable code, i.e. code that just needed to be upgradeable but not reusable. And so, in 2.x, upgradeability, and the ability to link to EVM packages were completely separated.
```

Comment:
>  We realized that these two contracts made sense when it came to EVM packages, but didn't when it came to parts of your project that weren't meant to be reusable code

This is true for Package and ImplementationProvider, but not for App. An App is not needed for sharing or reusing code; it only concentrates ownership of all proxies in a single address, different from the user's account. This has two benefits: 1) no issues when interacting with transparent proxies from the user's account, and 2) you can transfer ownership of the app to a multisig or another governance structure, to decentralize upgradeability control of all your contracts.

Reply:
After writing this, I realize that it could be somewhat confusing that `publish` has this twofold behaviour. We'll need to be particularly careful on how we explain it. And eventually, we could [remove one of those behaviours](https://github.com/zeppelinos/zos/issues/291) and just keep the _publishing reusable code_ part.